File: flask/app/models/toy.py
from sqlalchemy.orm import relationship
from .db import db

class Toy(db.Model):
  __tablename__ = 'toys'

  id = db.Column(db.Integer, primary_key = True)
  name = db.Column(db.String)
  cat_id = db.Column(db.ForeignKey('cats.id'))
  cat = relationship('Cat', back_populates='toys')
  toy_type_id = db.Column(db.ForeignKey('toy_types.id'))
  toy_type = relationship('ToyType', back_populates='toys')


  def to_dict(self, ):
    return {
      "id": self.id,
      "name": self.name,
      "cat_id": self.cat_id,
      "toy_type_id": self.toy_type_id,
      "toy_type": self.toy_type.name
    }

File: express/db/models/toy.js
'use strict';
module.exports = (sequelize, DataTypes) => {
  const Toy = sequelize.define('Toy', {
    name: DataTypes.STRING,
    toy_type_id: DataTypes.INTEGER,
    cat_id: DataTypes.INTEGER
  }, {});
  Toy.associate = function(models) {
    Toy.belongsTo(models.Cat, {foreignKey: 'cat_id' });
    Toy.belongsTo(models.ToyType, {foreignKey: 'toy_type_id' });
  };
  return Toy;
};
